### PR TITLE
Decompile battle_tv previously known as battle_link_817C95C

### PR DESCRIPTION
**Pull request by DizzyEggg**
Finally got off that ride.

### CODE REVIEW

File: src/battle_tv.c [hunk 1]
```diff
@@ -0,0 +1,1584 @@
...
+
+static bool8 sub_817E0B8(u16 stringId)
+{
+    s32 i = 0;
+
+    while (1)
+    {
+        if (sUnknown_0860A8A4[i] == stringId)
+            break;
+        i++;
```
Comment (Diegoisawesome) [inline src/battle_tv.c]: Isn't this a `do while (sUnknown_0860A8A4[i++] != 0xFFFF`

File: src/battle_tv.c [hunk 2]
```diff
@@ -0,0 +1,1584 @@
...
+static void TrySetBattleSeminarShow(void)
+{
+    s32 i;
+    s32 dmgByMove[4];
+    u16 powerOverride;
+    u16 currMoveSaved;
+
+    if (gBattleTypeFlags & (BATTLE_TYPE_DOUBLE | BATTLE_TYPE_LINK | BATTLE_TYPE_x2000000))
+        return;
+    if (GetBattlerSide(gBattlerAttacker) == B_SIDE_OPPONENT)
```
Comment (Diegoisawesome) [inline src/battle_tv.c]: if/else if

File: src/battle_tv.c [hunk 3]
```diff
@@ -0,0 +1,1584 @@
...
+    if (gBattleMoves[gBattleMons[gBattlerAttacker].moves[gMoveSelectionCursor[gBattlerAttacker]]].power == 0)
+        return;
+
+    i = 0;
+    currMoveSaved = gBattleMons[gBattlerAttacker].moves[gMoveSelectionCursor[gBattlerAttacker]];
+    while (1)
+    {
+        if (currMoveSaved == sVariableDmgMoves[i])
+            break;
+        i++;
```
Comment (Diegoisawesome) [inline src/battle_tv.c]: do while?

File: src/battle_tv.c [hunk 4]
```diff
@@ -0,0 +1,1584 @@
...
+    }
+    else
+    {
+        s32 i = 0;
+        while (1)
+        {
+            if (moveId == sVariableDmgMoves[i])
+                break;
+            i++;
+            if (sVariableDmgMoves[i] == 0xFFFF)
```
Comment (Diegoisawesome) [inline src/battle_tv.c]: Do while?

File: src/battle_tv.c [hunk 5]
```diff
@@ -0,0 +1,1584 @@
...
+
+    if (GetBattlerSide(battlerId) == B_SIDE_PLAYER)
+        party = gPlayerParty;
+    else
+        party = gEnemyParty;
+
+    i = 0;
+    while (1)
+    {
+        if (i >= 4)
```
Comment (Diegoisawesome) [inline src/battle_tv.c]: Definitely not a for loop?

Comment (DizzyEggg) [reply src/battle_tv.c]: definitely not

**Pull request merged**